src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -661,6 +669,57 @@ private void validateRenewalInfo(CompositePurchaseOrder compPO) {
       .thenApply(v -> compPO);
   }
 
+  private CompletableFuture<Void> validateFundDistributionTotal(CompositePurchaseOrder compPO) {
+    if (compPO.getWorkflowStatus() == WorkflowStatus.OPEN && compPO.getTotalEstimatedPrice() != null) {
+      Double totalEstimatedPrice = compPO.getTotalEstimatedPrice();
+      Double remainingPercent = 100.0;
+
+      for (CompositePoLine cPoLine : compPO.getCompositePoLines()) {
+        String currency = cPoLine.getCost().getCurrency();
+        MonetaryAmount remainingAmount = Money.of(totalEstimatedPrice, currency);
+
+        for (FundDistribution fundDistribution : cPoLine.getFundDistribution()) {
+          DistributionType dType = fundDistribution.getDistributionType();
+
+          if (dType == DistributionType.PERCENTAGE) {
+            Double percentValue = fundDistribution.getValue();
+            remainingPercent -= percentValue;
+
+            MonetaryAmount totalEstimatedPriceMoney = Money.of(totalEstimatedPrice, currency);
+            /**
+             * calculate remaining amount to carry forward, required if there are more fund distributions with percentage and amount
+             * currentAmount = totalEstimatedPrice * value/100;
+             * remainingAmount = totalEstimatedPrice - currentAmount;
+             */
+            MonetaryAmount currentAmount = totalEstimatedPriceMoney.with(MonetaryOperators.percent(percentValue));
+            remainingAmount = remainingAmount.subtract(currentAmount);
+
+            if (remainingPercent < 0) {
+              throw new HttpException(422, INCORRECT_FUND_DISTRIBUTION_TOTAL);
+            }
+          } else if (dType == DistributionType.AMOUNT) {
+            Double amountValue = fundDistribution.getValue();
+            MonetaryAmount amountValueMoney = Money.of(amountValue, currency);
+            remainingAmount = remainingAmount.subtract(amountValueMoney);
+
+            /**
+             * calculate remaining percentage to carry forward, required if there are more fund distributions with amount and percentage
+             * remainingPercentage = remainingAmount/totalEstimatedPrice * 100
+             */
+            MonetaryAmount divideResult = remainingAmount.divide(totalEstimatedPrice);
+            remainingPercent = divideResult.multiply(100).getNumber().doubleValue();
+
+            MonetaryAmount zeroMoney = Money.of(0, currency);
+            if (remainingAmount.isLessThan(zeroMoney)) {
```

Comment:
I would recommend choosing one of the validation's variants either percentage or amount and convert all distributions to the same type, it will reduce the amount of duplicated code.

Reply:
Done, converting all to the amount.